sdk/core/Azure.Core.Experimental/src/JsonData.cs

```diff
@@ -193,7 +193,7 @@ private void InitFromElement(JsonElement element)
         /// <param name="propertyName">The name of the property to get.</param>
         /// <returns>The value for a given property</returns>
         /// <remarks>If the <see cref="Kind"/> property is not <see cref="JsonValueKind.Object"/> this method throws <see cref="InvalidOperationException"/>.</remarks>
-        public T Get<T>(string propertyName)
+        public T? Get<T>(string propertyName)
```

Comment:
This probably makes sense, but why are we changing this here?

Reply:
Agreed - let's make these changes in a separate PR - do they need to ship this month?

Reply:
> This probably makes sense, but why are we changing this here?

.net 6.0 complains about this. So I need to either fix it or disable nullability checks for this file.

Reply:
Since this type is migrating to Core anyway, I think we should make the smallest change necessary and disable nullable checks for now.

Reply:
Well, that is somewhat dangerous as it introduces the risk that we forget to re-enable. I don't think we have tooling to check for this.

Reply:
I think the smallest _safe_ change here would be to just not target net6.0 in experimental yet.

Reply:
This type will disappear from the project, so the nullable change will go with it.

Reply:
Yes, but it will be copy pasted into Core, hence the risk.

Reply:
No - there is already a PR in progress.

Reply:
Ah ok - seems fine then.

Reply:
Disabling nullable causes more subsequent changes, so minimal change is indeed not adding .net 6.0 for now.